src/raft.c

```diff
@@ -1620,23 +1620,19 @@ void replaceShardGroups(RedisRaftCtx *rr, raft_entry_t *entry)
     ShardingInfo *si = rr->sharding_info;
 
     if (si->shard_group_map != NULL) {
+        void *data;
         RedisModuleDictIter *iter = RedisModule_DictIteratorStartC(si->shard_group_map, "^", NULL, 0);
 
-        char *key;
-        size_t key_len;
-        ShardGroup *data;
-
-        while ((key = RedisModule_DictNextC(iter, &key_len, (void **) &data)) != NULL) {
+        while (RedisModule_DictNextC(iter, NULL, &data) != NULL) {
             ShardGroupFree(data);
         }
         RedisModule_DictIteratorStop(iter);
         RedisModule_FreeDict(rr->ctx, si->shard_group_map);
-        si->shard_group_map = NULL;
```

Comment:
It's better to keep `si->shard_group_map = NULL;`, although in the current implementation we immediately set a new value, in order to avoid using freed memory in the future.

Reply:
sure